lib/tapioca/gem/listeners/source_location.rb

```diff
@@ -0,0 +1,54 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Tapioca
+  module Gem
+    module Listeners
+      class SourceLocation < Base
+        extend T::Sig
+
+        private
+
+        sig { override.params(event: ConstNodeAdded).void }
+        def on_const(event)
+          file, line = Object.const_source_location(event.symbol)
+          add_source_location_comment(event.node, file, line)
+        end
+
+        sig { override.params(event: ScopeNodeAdded).void }
+        def on_scope(event)
+          location = Runtime::Trackers::ConstantDefinition.locations_for(event.symbol)
```

Comment:
Why do we need to use the tracker to find the location here? Might be worth writing a short comment so anyone who revisits this in the future understands.

Reply:
Also this is a nitpick, but I found the next line confusing because this variable is called `location` -- it's an array with multiple locations, right?

Reply:
I added a comment. And yes, it's supposed to be `locations` in plural and then `location` after the `find`.